presto-main/src/main/java/com/facebook/presto/metadata/MetadataListing.java

```diff
@@ -73,6 +74,18 @@ private MetadataListing() {}
         return accessControl.filterTables(session.getRequiredTransactionId(), session.getIdentity(), prefix.getCatalogName(), tableNames);
     }
 
+    public static Set<GrantInfo> listTablePrivileges(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix, String grantee)
+    {
+/*        Set<PrivilegeInfo> grants = metadata.listTablePrivileges(session, prefix, grantee).stream()
```

Comment:
I don't lose it. That check is now redundant. I am extracting the grantee from the session via `extractGrantee(session, filters);` and passing that to `listTablePrivileges` method. So the privileges returned are only for the given session user.